keps/sig-node/2413-seccomp-by-default/README.md

```diff
@@ -0,0 +1,593 @@
+  <!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required _prior to targeting to a milestone / release_.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Enable seccomp by default for all workloads running on Kubernetes to improve the
+default security of the overall system.
+
+## Motivation
+
+Kubernetes provides a native way to specify seccomp profiles for workloads,
+which is disabled by default today. Seccomp adds a layer of security that could
+help prevent CVEs or 0-days if enabled by default. If we enable seccomp by
+default, we make implicitly Kubernetes more secure.
+
+### Goals
+
+Provide a way to enable seccomp by default for Kubernetes.
+
+### Non-Goals
+
+Everything else related to the feature.
+
+## Proposal
+
+We introduce a feature gate that enables a seccomp for all workloads by default.
+There are a few options for what should be the default seccomp profile.
+
+The most preferred solution is to promote the `RuntimeDefault` profile
+(previously the `runtime/default` annotation) to the new default one.
+
+Container runtimes already have their own defined default profile which gets
```

Comment:
I believe both containerd & crio allow you to customize the default seccomp profile. Do they both allow `unconfined` to be set as the default? If so, I think it's worth calling that out as a risk mitigation. In other words, if a user really wants to keep the unconfined default they can configure it at the runtime level.

Reply:
> I believe both containerd & crio allow you to customize the default seccomp profile. 

Did a small research and I think:
* containerd
  * can only use the default profile for `RuntimeDefault`
  * can use a different profile for empty (unconfined) workloads via `unset_seccomp_profile`
* CRI-O
  * can specify a different `RuntimeDefault` profile via `seccomp_profile`
  * can use `RuntimeDefault` for empty  (unconfined) workloads via `seccomp_use_default_when_empty`

> Do they both allow `unconfined` to be set as the default? If so, I think it's worth calling that out as a risk mitigation. 

I don't think so, this has not been covered by runtimes yet.

> In other words, if a user really wants to keep the unconfined default they can configure it at the runtime level.

A workaround could be to let users specify an "allow all syscalls" profile at the runtime level. This would work at least for CRI-O.